src/Compiler/Checking/ConstraintSolver.fs

```diff
@@ -1814,7 +1856,11 @@ and SolveMemberConstraint (csenv: ConstraintSolverEnv) ignoreUnresolvedOverload
                   else 
                       do! CheckMethInfoAttributes g m None minfo
                       return TTraitSolved (minfo, calledMeth.CalledTyArgs, calledMeth.OptionalStaticType)
-                          
+              | None, None, None, Some (ilfinfo, isSet) -> 
```

Comment:
So, using this logic here, traits are going to work for **either** records, anon records or fields. Not all of them together. Should work properly for **just** properties+fields though.

Something to test to.